lib/grape/middleware/logger.rb

```diff
@@ -3,79 +3,75 @@
-      #   Exceptions are logged as a 500 status and re-raised
-      #   Other "errors" are caught, logged and re-thrown
-      def call!(env)
-        @env = env
-        before
-        error = catch(:error) do
-          begin
-            @app_response = @app.call(@env)
-          rescue => e
-            after_exception(e)
-            raise e
-          end
-          nil
-        end
-        if error
-          after_failure(error)
-          throw(:error, error)
-        else
-          after(@app_response.status)
-        end
-        @app_response
+  # @note Error and exception handling are required for the +after+ hooks
+  #   Exceptions are logged as a 500 status and re-raised
+  #   Other "errors" are caught, logged and re-thrown
+  def call!(env)
+    @env = env
+    before
+    error = catch(:error) do
+      begin
+        @app_response = @app.call(@env)
+      rescue => e
+        after_exception(e)
+        raise e
       end
+      nil
+    end
+    if error
+      after_failure(error)
+      throw(:error, error)
+    else
+      after(@app_response.status)
+    end
+    @app_response
+  end
 
-      def after(status)
-        logger.info "Completed #{status} in #{((Time.now - start_time) * 1000).round(2)}ms"
-        logger.info ''
-      end
+  def after(status)
+    logger.info "Completed #{status} in #{((Time.now - start_time) * 1000).round(2)}ms"
+    logger.info ''
+  end
 
-      #
-      # Helpers
-      #
+  #
+  # Helpers
+  #
 
-      def after_exception(e)
-        logger.info %Q(  Error: #{e.message})
-        after(500)
-      end
+  def after_exception(e)
+    logger.info %Q(  Error: #{e.message})
+    after(500)
+  end
 
-      def after_failure(error)
-        logger.info %Q(  Error: #{error[:message]}) if error[:message]
-        after(error[:status])
-      end
+  def after_failure(error)
+    logger.info %Q(  Error: #{error[:message]}) if error[:message]
+    after(error[:status])
+  end
 
-      def parameters
-        request_params = env['grape.request.params'].to_hash
-        request_params.merge!(env['action_dispatch.request.request_parameters'] || {}) # for Rails
-        if @options[:filter]
-          @options[:filter].filter(request_params)
-        else
-          request_params
-        end
-      end
+  def parameters
+    request_params = env['grape.request.params'].to_hash
+    request_params.merge!(env['action_dispatch.request.request_parameters'] || {}) # for Rails
+    if @options[:filter]
+      @options[:filter].filter(request_params)
+    else
+      request_params
+    end
+  end
 
-      def start_time
-        @start_time ||= Time.now
-      end
+  def start_time
+    @start_time ||= Time.now
+  end
 
-      def logger
-        @logger ||= @options[:logger] || ::Logger.new(STDOUT)
-      end
-    end
+  def logger
+    @logger ||= @options[:logger]
+    @logger ||= defined?(Rails) && Rails.logger.present? ? Rails.logger : Logger.new(STDOUT)
   end
```

Comment:
@ridiculous I think it would be nice. When coding to open the pull-request my goal was to not change much, but if you're open to suggestions we could really refactor the method. Yet first let us finish this feature. :dancer: